internal/checkers/encoded_compare.go

```diff
@@ -50,23 +50,32 @@ func (checker EncodedCompare) Check(pass *analysis.Pass, call *CallMeta) *analys
 		proposed = "JSONEq"
 	case isYAMLStyleExpr(pass, a), isYAMLStyleExpr(pass, b):
 		proposed = "YAMLEq"
+	default:
+		return nil
 	}
 
-	if proposed != "" {
-		return newUseFunctionDiagnostic(checker.Name(), call, proposed,
-			analysis.TextEdit{
-				Pos:     lhs.Pos(),
-				End:     lhs.End(),
-				NewText: formatWithStringCastForBytes(pass, a),
-			},
-			analysis.TextEdit{
-				Pos:     rhs.Pos(),
-				End:     rhs.End(),
-				NewText: formatWithStringCastForBytes(pass, b),
-			},
-		)
+	formattedA, ok := formatToString(pass, a)
+	if !ok {
+		return nil
 	}
-	return nil
+
+	formattedB, ok := formatToString(pass, b)
```

Comment:
I don't see positive test cases for this, new cases in `// Ignored.` section only

Reply:
This PR is old. I will have to read again what I wanted to do.

Thanks for the review